src/App.tsx

```diff
@@ -15,47 +16,123 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+export function getReorderedGoods(
+  goods: string[], sortType: SortType, isReversed: boolean,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((good1, good2) => {
+    if (sortType === SortType.ALPHABET) {
+      return good1.localeCompare(good2);
+    }
+
+    if (sortType === SortType.LENGTH) {
+      return good1.length - good2.length;
+    }
+
+    return 0;
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  const renderedGoods = visibleGoods.map((good) => (
+    <li
+      data-cy="Good"
+      key={good}
+    >
+      {good}
+    </li>
+  ));
+
+  return renderedGoods;
+}
+
 export const App: React.FC = () => {
+  const [reversed, setReversed] = useState(false);
+
+  const reverse = () => {
+    setReversed(current => !current);
+  };
+
+  const [sorted, setSort] = useState(SortType.NONE);
+
+  const sortByLength = () => {
+    setSort(SortType.LENGTH);
+  };
+
+  const sortAlphabetically = () => {
+    setSort(SortType.ALPHABET);
+  };
+
+  const reset = () => {
+    setReversed(false);
+    setSort(SortType.NONE);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn(
+            'button',
+            'is-success',
+            { 'is-light': sorted !== SortType.ALPHABET },
+          )}
+          onClick={() => sortAlphabetically()}
```

Comment:
```suggestion
          onClick={sortAlphabetically}
```
**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**